### PR TITLE
Set the current directory of the server to the directory containing the executable

### LINKED ISSUE #722: File handle to root dir left open
For my sins I'm building rust in a conda environment. That environment's directory is renamed at the end of the build.

But fails to be renamed as sscache has an open handle to the dir in question: 
C:\conda-bld\my-proj_1586446013252\work

It would be really great if it didn't keep file handles open. You can see it in process explorer that it's kept a handle on the build dir open.

### PR DESCRIPTION
**Pull request by glandium**
On Windows, files and directories can't be modified if they are "locked" by some process that has them opened. When the server is spawned on the first compile, the directory it was launched from is kept open as its current directory, which prevents some actions on the directory later on (like removing it or renaming it), as long as the server keeps running.

One directory that is less likely to be renamed or removed during the lifetime of the server is the directory containing sccache itself, so we use that as sccache's current directory instead.

Fixes #722

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -209,6 +211,13 @@ fn run_server_process() -> Result<ServerStartup> {
         v.push(0);
         v
     };
+    let workdir = exe_path
+        .parent()
+        .expect("executable path has no parent?!")
+        .as_os_str()
+        .encode_wide()
+        .chain(Some(0u16))
```
Comment (mitchhentges) [inline src/commands.rs:219]: Hmm, I'm surprised that there isn't a better existing abstraction that we can use here that encapsulates these technical details. Something like:
```
let workdir: WindowsStringParameterThinger = exe_path
    .parent()
    .expect(...)
    .as_os_str()
    .into::<WindowsStringParameterThinger>();

// ...

CreateProcessW(
    ...,
    workdir.to_ptr()
    ...,
)
```

**Review by mitchhentges — APPROVED** [review]

**Review by cgsheeh — APPROVED** [review]
Looks good to me - guessing we can't avoid taking the lock on the directory altogether as the issue reporter suggests?

**Comment by glandium** [comment]
I don't think it's possible to close the handle for the current directory, no.

File: src/commands.rs (hunk 1 above)
Comment (glandium) [reply src/commands.rs:219]: There is https://crates.io/crates/widestring but we're not currently using it. I just used the same pattern as already in use in the same file.

**Pull request merged**